Pixman build fails

Summary
Building gnuplot on macOS High Sierra, the build of dependency pixman failed due the removal of __builtin_shuffle from clang 5.0, as reported here: https://bugs.freedesktop.org/show_bug.cgi?id=104886. I've manually applied the patch given there, and then pixman built.

### Expected Result
Build pixman.

### Actual Result
Build failed with:
```console
==> Installing pixman
==> Using cached archive: /Applications/Spack/spack/var/spack/cache/pixman/pixman-0.34.0.tar.gz
==> Staging archive: /Applications/Spack/spack/var/spack/stage/pixman-0.34.0-2rhnonvnl6rshlcan4jgpfdiwo2a3avg/pixman-0.34.0.tar.gz
==> Created stage in /Applications/Spack/spack/var/spack/stage/pixman-0.34.0-2rhnonvnl6rshlcan4jgpfdiwo2a3avg
==> No patches needed for pixman
==> Building pixman [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j4'

5 errors found in build log:
     595      CC       infinite-loop.o
     596      CC       trap-crasher.o
     597      CC       fence-image-self-test.o
     598      CC       region-translate-test.o
     599      CC       fetch-test.o
     600      CC       a1-trap-test.o
   601    utils-prng.c:207:27: error: use of unknown builtin '__builtin_shuffle' [-Wimplicit-function-declaration]
     602                randdata.vb = __builtin_shuffle (randdata.vb, bswap_shufflemask);
     603                              ^
   604    utils-prng.c:207:25: error: assigning to 'uint8x16' (vector of 16 'uint8_t' values) from incompatible type 'int'
     605                randdata.vb = __builtin_shuffle (randdata.vb, bswap_shufflemask);
     606                            ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     607    2 errors generated.
   608    make[2]: *** [utils-prng.lo] Error 1
     609    make[2]: *** Waiting for unfinished jobs....
   610    make[1]: *** [all-recursive] Error 1
   611    make: *** [all] Error 2
```

### Steps to reproduce the issue

```console
$ spack install gnuplot
```

### Information on your system
macOS High Sierra 10.13.4
```console
$ clang --version
Apple LLVM version 9.1.0 (clang-902.0.39.1)
Target: x86_64-apple-darwin17.5.0
Thread model: posix
```
### Solution
Apply patch https://bugs.freedesktop.org/show_bug.cgi?id=104886
-----

--- a/var/spack/repos/builtin/packages/pixman/clang.patch
+++ b/var/spack/repos/builtin/packages/pixman/clang.patch
@@ -1,0 +1,28 @@
+--- pixman-0.34.0/test/utils-prng.c.bak	2018-02-01 00:34:15.757691302 +0100
++++ pixman-0.34.0/test/utils-prng.c	2018-02-01 00:36:38.285690995 +0100
+@@ -199,12 +199,24 @@
+         }
+         else
+         {
++
++#ifndef __has_builtin
++#define __has_builtin(x) 0
++#endif
++
+ #ifdef HAVE_GCC_VECTOR_EXTENSIONS
++# if !defined(__clang__) || __has_builtin(__builtin_shuffle)
+             const uint8x16 bswap_shufflemask =
+             {
+                 3, 2, 1, 0, 7, 6, 5, 4, 11, 10, 9, 8, 15, 14, 13, 12
+             };
+             randdata.vb = __builtin_shuffle (randdata.vb, bswap_shufflemask);
++# elif __has_builtin(__builtin_shufflevector)
++            randdata.vb = __builtin_shufflevector(randdata.vb, randdata.vb, 3, 2, 1, 0, 7, 6, 5, 4, 11, 10, 9, 8, 15, 14, 13, 12);
++# else
++#  error "GCC_VECTOR_EXTENSION is defined, but neither __builtin_shuffle nor __builtin_shufflevector is supported!"
++# endif
++
+             store_rand_128_data (buf, &randdata, aligned);
+             buf += 16;
+ #else
+

--- a/var/spack/repos/builtin/packages/pixman/package.py
+++ b/var/spack/repos/builtin/packages/pixman/package.py
@@ -40,6 +40,13 @@ class Pixman(AutotoolsPackage):
     depends_on('pkgconfig', type='build')
     depends_on('libpng')
 
+    # As discussed here:
+    # https://bugs.freedesktop.org/show_bug.cgi?id=104886
+    # __builtin_shuffle was removed in clang 5.0.
+    # From version 9.1 apple-clang is based on clang 5.0.
+    # Patch is obtained from above link.
+    patch('clang.patch', when='%clang@9.1.0-apple:')
+
     def configure_args(self):
         args = [
             '--enable-libpng',

